### PR TITLE
[Tag] Improve outline behavior

### PR DESCRIPTION
**Pull request by llorca**
- Add `tabIndex` on interactive tags only, for proper keyboard navigation
- Clean up focus styles

### CODE REVIEW

**Comment by giladgray** [comment]
@llorca `TagInput` backspace & arrow key interactions look broken -- no focus state

File: packages/core/src/components/tag-input/_tag-input.scss [hunk 1]
```diff
@@ -17,34 +21,47 @@ $ti-icon-padding-large: ($pt-input-height-large - $pt-icon-size-large) / 2;
...
     }
   }
 
   .#{$ns}-tag {
+    position: relative;
     // NOTE: in order to wrap long words, you must set explicit width on TagInput,
     // or use .#{$ns}-fill CSS class modifier.
     overflow-wrap: break-word;
+
+    &::before {
```
Comment (giladgray) [inline packages/core/src/components/tag-input/_tag-input.scss]: not a fan of this special case, and even less a fan of not simply using `outline` / the default focus styles (which it used to before this PR)

File: packages/core/src/components/tag/tag.tsx [hunk 2]
```diff
@@ -88,7 +88,7 @@ export class Tag extends React.PureComponent<ITagProps, {}> {
         ) : null;
 
         return (
-            <span {...htmlProps} className={tagClasses}>
+            <span {...htmlProps} className={tagClasses} tabIndex={interactive ? 0 : -1}>
```
Comment (giladgray) [inline packages/core/src/components/tag/tag.tsx]: i think we want `undefined` in the false case (emphasis mine):

> A negative value (usually `tabindex="-1"`) means that the element **should be focusable, but should not be reachable** via sequential keyboard navigation. Mostly useful to create accessible widgets with JavaScript.
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex

File: packages/core/src/components/tag-input/_tag-input.scss [hunk 3]
```diff
@@ -17,34 +21,39 @@ $ti-icon-padding-large: ($pt-input-height-large - $pt-icon-size-large) / 2;
...
+    margin-right: $tag-input-icon-padding;
 
     > * {
-      margin-bottom: $ti-padding;
+      margin-bottom: $tag-input-padding;
     }
   }
 
   .#{$ns}-tag {
+    position: relative;
```
Comment (giladgray) [inline packages/core/src/components/tag-input/_tag-input.scss]: don't think this chance is necessary. leftover from previous attempt?

File: packages/core/src/components/tag-input/_tag-input.scss [hunk 4]
```diff
@@ -17,34 +21,39 @@ $ti-icon-padding-large: ($pt-input-height-large - $pt-icon-size-large) / 2;
...
     // NOTE: in order to wrap long words, you must set explicit width on TagInput,
     // or use .#{$ns}-fill CSS class modifier.
     overflow-wrap: break-word;
+
+    &.#{$ns}-active {
+      @include focus-outline(0);
```
Comment (giladgray) [inline packages/core/src/components/tag-input/_tag-input.scss]: i kinda think we want to adjust outline offset on _all_ tags, not just in tag inputs:

image
(note how the outline touches the neighboring tags)

image

so this change should move back to core tag styles.

**Comment by giladgray** [comment]
@llorca please merge latest master for the new build steps

File: packages/core/src/components/tag-input/_tag-input.scss (hunk 4 above)
Comment (llorca) [reply packages/core/src/components/tag-input/_tag-input.scss]: I don't think that's the right move, but did it anyway, Not sure what to do with outline design in general

File: packages/core/src/components/tag-input/_tag-input.scss [hunk 5]
```diff
@@ -5,10 +5,14 @@
...
 
-$ti-padding: ($pt-input-height - $tag-height) / 2;
+$tag-input-padding: ($pt-input-height - $tag-height) / 2;
 
-$ti-icon-padding: ($pt-input-height - $pt-icon-size-standard) / 2;
-$ti-icon-padding-large: ($pt-input-height-large - $pt-icon-size-large) / 2;
+$tag-input-icon-padding: ($pt-input-height - $pt-icon-size-standard) / 2;
+$tag-input-icon-padding-large: ($pt-input-height-large - $pt-icon-size-large) / 2;
+
+$tag-input-outline-color: $pt-outline-color;
```
Comment (giladgray) [inline packages/core/src/components/tag-input/_tag-input.scss]: remove this variable, use original name directly

File: packages/core/src/components/tag-input/_tag-input.scss [hunk 6]
```diff
@@ -5,10 +5,14 @@
...
+$tag-input-outline-offset: 1px;
+$tag-input-outline-size: 2px;
```
Comment (giladgray) [inline packages/core/src/components/tag-input/_tag-input.scss]: these two vars are unused; please remove

File: packages/core/src/components/tag/_common.scss [hunk 7]
```diff
@@ -120,6 +121,7 @@ $tag-padding-large: ($tag-height-large - $pt-icon-size-large) / 2 !default;
       background-color: rgba($background-color, $opacity + 0.1);
     }
 
+    &.pt-active,
```
Comment (giladgray) [inline packages/core/src/components/tag/_common.scss]: `#{$ns}`

File: packages/core/src/components/tag/_common.scss [hunk 8]
```diff
@@ -106,6 +106,7 @@ $tag-padding-large: ($tag-height-large - $pt-icon-size-large) / 2 !default;
       background-color: rgba($background-color, $opacity - 0.15);
     }
 
+    &.pt-active,
```
Comment (giladgray) [inline packages/core/src/components/tag/_common.scss]: `pt` &rArr; `#{$ns}`

File: packages/core/src/components/tag-input/_tag-input.scss [hunk 9]
```diff
@@ -5,10 +5,14 @@
 @import "../forms/common";
 @import "../tag/common";
 
-$ti-padding: ($pt-input-height - $tag-height) / 2;
+$tag-input-padding: ($pt-input-height - $tag-height) / 2;
 
-$ti-icon-padding: ($pt-input-height - $pt-icon-size-standard) / 2;
-$ti-icon-padding-large: ($pt-input-height-large - $pt-icon-size-large) / 2;
+$tag-input-icon-padding: ($pt-input-height - $pt-icon-size-standard) / 2;
+$tag-input-icon-padding-large: ($pt-input-height-large - $pt-icon-size-large) / 2;
```
Comment (giladgray) [inline packages/core/src/components/tag-input/_tag-input.scss]: can you add `!default` to all these (surviving) vars please?

File: packages/core/src/components/tag/_common.scss [hunk 10]
```diff
@@ -27,6 +27,10 @@ $tag-padding-large: ($tag-height-large - $pt-icon-size-large) / 2 !default;
   color: $pt-dark-text-color;
   font-size: $pt-font-size-small;
 
+  &.#{$ns}-active {
```
Comment (giladgray) [inline packages/core/src/components/tag/_common.scss]: ah dang actually i think this should be `&:focus` here, just to adjust the outline-offset.

and bring back this selector in `.pt-tag-input .pt-tag` like you had earlier

😄

Comment (llorca) [reply packages/core/src/components/tag/_common.scss]: pushed another commit, hopefully I got it right this time!

**Pull request merged**